tests/miral/runner.cpp

```diff
@@ -73,6 +106,66 @@ TEST_F(Runner, x11_socket_is_not_returned_by_default)
     miral::TestServer::invoke_runner([](auto& runner){ EXPECT_FALSE(runner.x11_display().is_set()); });
 }
 
+TEST_F(Runner, register_signal_handler_before_setup_invokes_callback_after_setup)
+{
+    int const signum{SIGUSR1};
+
+    wait_for_locking_callback();
+    register_signal_handler({signum}, [this](int){ locking_callback(); });
+
+    miral::TestServer::SetUp();
+
+    EXPECT_CALL(*this, callback())
+        .Times(AtLeast(1));
+
+    kill(getpid(), signum);
+    wait_to_tear_down();
+}
```

Comment:
Or... we could use a naked semaphore (a shame that latch isn't available for 20.04):

```suggestion
TEST_F(Runner, register_signal_handler_before_setup_invokes_callback_after_setup)
{
    std::binary_semaphore semaphore{0};
    int const signum{SIGUSR1};

    register_signal_handler({signum}, [this, &](int){ locking_callback(); semaphore.release(); });

    miral::TestServer::SetUp();

    EXPECT_CALL(*this, callback())
        .Times(AtLeast(1));

    kill(getpid(), signum);
    semaphore.try_acquire_until(/* a short wait*/); // Wait for callback before exiting scope
}
```